File: Api/Services/OrderService.cs
using Api.Abstractions;
using Api.Models;

namespace Api.Services;

public class OrderService : IOrderService
{
    private readonly IOrderRepository _orderRepository;

    public OrderService(IOrderRepository orderRepository)
    {
        _orderRepository = orderRepository;
    }

    public List<Order> GetOrdersByCompanyId(int companyId)
    {
        return _orderRepository.GetOrdersByCompanyId(companyId);
    }
}

File: web-app/src/app/services/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Order } from '../models/order';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private http: HttpClient) { }

  getOrdersByCompanyId(companyId: number): Observable<Order[]> {
    const url = `${environment.ordersBaseUrl}/Order/Company/${companyId}`;
    return this.http.get<Order[]>(url);
  }
}
